### PR TITLE
chore: upgrade to 6.1.0

### PR DESCRIPTION
**Pull request by sighphyre**
This brings the client up to date with Unleash enterprise version 6.1.0. To do that there are a few subtle but important changes:

### Removed UI config

UI config includes two fields: `feature_enabled` and `featureEnabled` which do not play nicely with the OpenAPI generator. This is a rapidly changing API and probably doesn't belong in this project. This may have effects on the downstream terraform plugin, since that project uses the version number to emit warnings. That's probably just a sacrifice that's going to happen

### AllowAdditionalProperties in generating the client

More specifically, this turns off `disallowAdditionalPropertiesIfNotPresent` in the generator script. More details [here](https://openapi-generator.tech/docs/generators/go). This changes the behavior in the generated client to tolerate additional unknown properties being sent back in the JSON payloads. This is necessary because the `accountType` property on users is not specified in the spec when returned via the GetProjectAccess API

### Fix the project tests

There's a single case here where creating a project would fail because the spec incorrectly marks the project id as deprecated and no longer necessary. This causes the client to not set the field in it's constructor. We work around that by setting it manually.

### Fix the user tests

They weren't broken, they just sucked. They were all stateful meaning if something in the suite failed, the feedback loop required you to tear down and respin the database. These now clean up after themselves, meaning you can run tests as many times as you want. Yay!

# How to deal with this PR

Ignore everything in the /client folder and the README. That's all autogenned. 

All of the salient changes are commented. Start by looking at those

### CODE REVIEW

File: generate.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ openapi-generator-cli generate \
     --openapi-normalizer REFACTOR_ALLOF_WITH_PROPERTIES_ONLY=true \
     --openapi-normalizer RESOLVE_INLINE_ENUMS=true \
     --openapi-normalizer REF_AS_PARENT_IN_ALLOF=true \
-    --additional-properties packageName=client \
+    --additional-properties packageName=client,disallowAdditionalPropertiesIfNotPresent=false \
```
Comment (sighphyre) [inline generate.sh:20]: Salient change to tolerate additional JSON properties

File: test/api_projects_test.go [hunk 2]
```diff
@@ -21,7 +22,8 @@ func Test_client_ProjectsAPIService(t *testing.T) {
 
 	t.Run("Test ProjectsAPIService CreateProject", func(t *testing.T) {
 		if enterpriseEnvironmentAvailable() {
-			createProjectSchema := *client.NewCreateProjectSchema("pet-shop-project", "Pet shop project")
+			createProjectSchema := *client.NewCreateProjectSchema("pet-shop-project")
+			createProjectSchema.SetId("pet-shop-project")
```
Comment (sighphyre) [inline test/api_projects_test.go:26]: Salient change, this fixes the default project creation failing

File: test/api_projects_test.go [hunk 3]
```diff
@@ -79,6 +81,8 @@ func Test_client_ProjectsAPIService(t *testing.T) {
 
 			resp, httpRes, err := apiClient.ProjectsAPI.GetProjectAccess(context.Background(), projectId).Execute()
 
+			fmt.Println(err)
```
Comment (sighphyre) [inline test/api_projects_test.go:84]: Not salient but gosh darn does it making debugging this test easier. I'd like to keep it

File: test/api_users_test.go [hunk 4]
```diff
@@ -76,13 +76,28 @@ func createRole(name string, description string, roleType string) *client.Create
 	return newRole
 }
```
Comment (sighphyre) [inline test/api_users_test.go:77]: This file is a rewrite of the tests. Could use eyes here

File: operations.yaml [hunk 5]
```diff
@@ -27,11 +27,9 @@ inverseOperationIds:
...
   - setProjectAccess
-  
+
   - createRole
   - getRoles
   - getRoleById
   - updateRole
-  - deleteRole
-  
-  - getUiConfig
```
Comment (sighphyre) [inline operations.yaml:37]: Salient change, we no longer include this api

**Review by gastonfournier — APPROVED** [review]
Looks good to me!

File: test/api_users_test.go [hunk 6]
```diff
@@ -76,13 +76,28 @@ func createRole(name string, description string, roleType string) *client.Create
...
+
 func Test_client_UsersAPIService(t *testing.T) {
 	apiClient := testClient()
 
 	t.Run("Test UsersAPIService CreateUser", func(t *testing.T) {
-
-		createUser(t, apiClient, "test")
-
+		user := createUser(t, apiClient, "test")
+		defer clean_up_user(user, apiClient)
```
[the remark this answers is not in the document]
Comment (sighphyre) [reply test/api_users_test.go:100]: Yuh! It's one of the really cool things in Golang!

**Pull request merged**